File: Server/src/main/java/core/net/packet/in/RunScriptPacketHandler.java
package core.net.packet.in;

import static api.ContentAPIKt.*;
import core.game.node.entity.player.Player;
import core.game.node.entity.player.link.RunScript;
import core.net.packet.IncomingPacket;
import core.net.packet.IoBuffer;
import core.tools.StringUtils;
import kotlin.jvm.functions.Function1;

/**
 * Represents the incoming packet to handle a run script.
 * @author 'Vexia
 */
public class RunScriptPacketHandler implements IncomingPacket {

	@Override
	public void decode(Player player, int opcode, IoBuffer buffer) {
		final Function1 script = player.getAttribute("runscript", null);
		if (script == null || player.getLocks().isInteractionLocked()) {
			return;
		}
		Object value = "";
		if (opcode == 244) {
			value = StringUtils.longToString(buffer.getLong());
		} else if (opcode == 23) {
			value = buffer.getInt();
		} else if (opcode == 65){
			value = buffer.getString();//"longInput"
		} else {
			value = buffer.getInt();
		}
		try {
			script.invoke(value);
		} catch (NumberFormatException nfe){
			sendDialogue(player, "That number's a bit large, don't you think?");
		}
		player.removeAttribute("runscript");
	}
}

File: Server/src/main/kotlin/rs09/net/packet/in/RunScriptPacketHandler.kt
package rs09.net.packet.`in`

import api.sendDialogue
import core.game.node.entity.player.Player
import core.net.packet.IncomingPacket
import core.net.packet.IoBuffer
import core.tools.StringUtils

/**
 * Handles an incoming script execution request packet.
 *
 * @author vddCore
 */
class RunScriptPacketHandler : IncomingPacket {

    private fun processInput(player: Player, value: Any, script: ((Any) -> Boolean)) {
        if (value is Int && value <= 0) return

        var input = value

        if (player.getAttribute("parseamount", false)) {
            input = value.toString().lowercase()

            if (!input.matches(Regex("^(\\d+)(k+|m+)?$"))) {
                sendDialogue(player, "That doesn't look right. Please try again.")
                return
            }

            input = input.replace("k", "000").replace("m", "000000")
        }

        try {
            script(input)
        } catch (_: NumberFormatException) {
            sendDialogue(player, "That number's a bit large, don't you think?")
        }
    }

    override fun decode(player: Player, opcode: Int, buffer: IoBuffer) {
        val script: ((Any) -> Boolean)? = player.getAttribute("runscript", null)

        if (script == null || player.locks.isInteractionLocked)
            return

        val value: Any = when(opcode) {
            244 -> StringUtils.longToString(buffer.long)
            65 -> buffer.string
            else -> buffer.int // Handles OpCode 23 and other cases.
        }

        processInput(player, value, script)

        player.removeAttribute("parseamount")
        player.removeAttribute("runscript")
    }
}
